buildSrc/src/main/groovy/org/elasticsearch/gradle/plugin/PluginBuildPlugin.groovy

```diff
@@ -115,6 +115,11 @@ class PluginBuildPlugin implements Plugin<Project> {
             if (isModule == false || isXPackModule) {
                 addNoticeGeneration(project, extension1)
             }
+            if (project.pluginManager.hasPlugin("elasticsearch.yaml-rest-test")
+                || project.pluginManager.hasPlugin("elasticsearch.java-rest-test")) {
+                //disable integTest task if project has been converted to use yaml or java rest test plugin
+                project.tasks.integTest.enabled = false
```

Comment:
So this doesn't actually _remove_ this task, we simply disable it in a central location to remove noise from build scripts. What's keeping us from actually getting rid of this altogether?

Reply:
> So this doesn't actually remove this task, we simply disable it in a central location to remove noise from build scripts.

correct.

> What's keeping us from actually getting rid of this altogether?

This PR only covers modules.  I am in the process of converting all of the projects to this pattern. Once that is done, I can (and will) remove `integTest` from here.